### PR TITLE
[Merged by Bors] - Always require auth header in VC

### LINKED ISSUE #2512: VC: requests may not contain Authorization header with API token
## Description

According to the [Validator Client API documentation](https://lighthouse-book.sigmaprime.io/api-vc-endpoints.html#post-lighthousevalidators), all requests (GET, POST, PATCH) must contain Authorization header with the token.

The current implementation only requires the API token in GET requests only, POST/PATCH requests will be processed even if they don't contain the Authorization header.

https://github.com/sigp/lighthouse/blob/stable/validator_client/src/http_api/mod.rs#L480

image

cc: @Nocasis

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

- Resolves #2512 

## Proposed Changes

Enforces that all routes require an auth token for the VC.

## TODO

- [x] Tests

### CODE REVIEW

File: validator_client/src/http_api/tests.rs [hunk 1]
```diff
@@ -455,6 +493,76 @@ fn invalid_pubkey() {
...
 }
 
+#[test]
+fn routes_with_invalid_auth() {
+    let runtime = build_runtime();
+    let weak_runtime = Arc::downgrade(&runtime);
+    runtime.block_on(async {
+        ApiTester::new(weak_runtime)
+            .await
+            .test_with_invalid_auth(|client| async move { client.get_lighthouse_version().await })
```
Comment (michaelsproul) [inline validator_client/src/http_api/tests.rs:503]: Just a style thing, but I think we can avoid the `async`/`await` wrapping

```suggestion
            .test_with_invalid_auth(|client| client.get_lighthouse_version())
```

**Review by michaelsproul — APPROVED** [review]
Looks good! Nice work!

File: validator_client/src/http_api/tests.rs (hunk 1 above)
Comment (paulhauner) [reply validator_client/src/http_api/tests.rs:503]: As discussed, I couldn't make this guy work :(

**Pull request closed without merging**